src/transformers/training_args.py

```diff
@@ -551,8 +549,18 @@ def __post_init__(self):
             )
 
         if isinstance(self.sharded_ddp, bool):
-            self.sharded_ddp = "simple" if self.sharded_ddp else "no"
-        self.sharded_ddp = ShardedDDPType(self.sharded_ddp)
+            self.sharded_ddp = "simple" if self.sharded_ddp else ""
+        if isinstance(self.sharded_ddp, str):
+            self.sharded_ddp = [ShardedDDPOption(s) for s in self.sharded_ddp.split()]
+        if self.sharded_ddp == [ShardedDDPOption.OFFLOAD]:
+            raise ValueError(
+                "`--sharded_ddp offload` can't work on its own. It needs to be added to `--sharded_ddp zero_dp_2` or "
+                "`--sharded_ddp zero_dp_3`"
+            )
```

Comment:
```suggestion
        if self.sharded_ddp == [ShardedDDPOption.OFFLOAD]:
            raise ValueError(
                "`--sharded_ddp offload` can't work on its own. It needs to be added to `--sharded_ddp zero_dp_2` or "
                "`--sharded_ddp zero_dp_3`". For example, `--sharded_ddp "zero_dp_2 offload"`.
            )
```

Let's help the user here a bit.